src/storage/irc_table_entry.cpp

```diff
@@ -54,67 +53,51 @@ TableFunction ICTableEntry::GetScanFunction(ClientContext &context, unique_ptr<F
 	auto table_credentials = IRCAPI::GetTableCredentials(context, ic_catalog, table_data->schema_name, table_data->name, ic_catalog.credentials);
 	CreateSecretInfo info(OnCreateConflict::REPLACE_ON_CONFLICT, SecretPersistType::TEMPORARY);
 	// First check if table credentials are set (possible the IC catalog does not return credentials)
-	if (!table_credentials.key_id.empty()) {
-		// Inject secret into secret manager scoped to this path
-		info.name = "__internal_ic_" + table_data->table_id;
-		info.type = "s3";
-		info.provider = "config";
-		info.storage_type = "memory";
-		info.options = {
-			{"key_id", table_credentials.key_id},
-			{"secret", table_credentials.secret},
-			{"session_token", table_credentials.session_token},
-			{"region", table_credentials.region},
-		};
+	for (auto &info : table_credentials.storage_credentials) {
+		if (info.name == "PLACEHOLDER") {
+			//! Set a name for the initial secret created from the LoadTableResult's 'config' object
+			if (StringUtil::StartsWith(ic_catalog.host, "s3tables")) {
+				info.name = "__internal_ic_" + table_data->table_id + "__" + table_data->schema_name + "__" + table_data->name;
+			} else {
+				info.name = "__internal_ic_" + table_data->table_id;
+			}
+		}
+
+		//! Limit the scope to the metadata location if no explicit scope was set
+		if (info.scope.empty()) {
+			std::string lc_storage_location;
+			lc_storage_location.resize(table_data->storage_location.size());
+			std::transform(table_data->storage_location.begin(), table_data->storage_location.end(), lc_storage_location.begin(), ::tolower);
+			size_t metadata_pos = lc_storage_location.find("metadata");
+			if (metadata_pos != std::string::npos) {
+				info.scope = {lc_storage_location.substr(0, metadata_pos)};
```

Comment:
This looks like it will have a bigger scope than the "scoped" secrets we create from the StorageCredential items, which might cause it to override the scoped ones.

Reply:
Solved by only creating this if there are no `storage-credentials` provided (as those will have their own specific scope already)